Fixes and pathing improvements

someaddons opened the pull request:
Closes #6100
Closes #6193
Closes #6198
# Changes proposed in this pull request:

Pathing now also maps corner nodes when going upstairs or dropping down, giving our citizen a more smooth path to follow and fixing some corner cases, like in 6100 where without that node they jump down from a ladder since the next node is diagonal in the opposite direction on the next floor.

Pathing now considers going down through passable blocks, which previously did not work in cases where the passable block is also of walkable ground type, like trapdoors are.

Improved pathfollow, it is now better at recognizing arrival at a path point and also displays this during debug draw, reached nodes are displayed as orange:
![image](https://user-images.githubusercontent.com/38401808/101158386-ee091b80-362b-11eb-87d7-5e242c747151.png)

Adjust min raidlevel before the first raid to be a bit lower
Fix pathing trying to drop down from places where the previous block is passable and dropable, e.g. jumping down vines instead of going them down.
Fix random path generation beeing able to end up on blocks a citizen cannot stand upon.
Fix guards doing stuck teleport during combat
Fix guards using rails during combat
Fix visitor zero position check throwing errors on clientside, when some other mods are creating one at 0 0 for 
rendering
Fix raids not spawning at low levels
Fix AI getting stuck when removing tasks
Fix citizen loading the colony to find the team, when not needed to do so

Review please

someaddons committed:
Pathing and other fixes

someaddons committed:
fix drop not beeing allowed when dropping

Raycoms commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java:
```diff
@@ -438,6 +440,11 @@ private IAIState checkAndAttackTarget()
         if (fighttimer > 0)
         {
             fighttimer--;
+            if (fighttimer == 0)
```
I sthis enough to have this here, what happens if combat finishes and they are hungry, or try to retreat with low health?

someaddons replied:
ye because this is the point which also prevents the interruption of the other AI's

Raycoms replied:
kk

Raycoms commented:
:shipit: 

[![Approved with PullApprove](https://img.shields.io/badge/maintainers-approved-brightgreen.svg)](https://pullapprove.com/ldtteam/minecolonies/pull-request/6196/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ldtteam/minecolonies)

Aduis1210 commented:
Will this fix the issue we are having with Guards not supporting the colony?

someaddons committed:
Fix raids not spawning at low levels
Fix AI getting stuck when removing tasks
Fix citizen loading the colony to find the team, when not needed to do so

Raycoms commented:
@Aduis1210 yes

marchermans commented on src/api/java/com/minecolonies/api/entity/ai/statemachine/basestatemachine/BasicStateMachine.java:
```diff
@@ -92,15 +92,11 @@ public void removeTransition(final T transition)
-            final List<T> temp = new ArrayList<>(eventTransitionMap.get(((IStateMachineEvent<?>) transition).getEventType()));
-            temp.remove(transition);
-            eventTransitionMap.put(((IStateMachineEvent<?>) transition).getEventType(), temp);
+            eventTransitionMap.get(((IStateMachineEvent<?>) transition).getEventType()).removeIf(t -> t == transition);
```
Are we sure this can not produce an NPE?
Would a getOrDefault(..., Collections.emptyList()) not be better here just to be sure.

someaddons replied:
ye it would indeed cause and NPE, thought that too when writing it and just left it to crash as sth must be wrong when trying to remove one that isnt added at all

marchermans commented on src/api/java/com/minecolonies/api/entity/ai/statemachine/basestatemachine/BasicStateMachine.java:
```diff
@@ -92,15 +92,11 @@ public void removeTransition(final T transition)
-            final List<T> temp = new ArrayList<>(transitionMap.get(transition.getState()));
-            temp.remove(transition);
-            transitionMap.put(transition.getState(), temp);
+            transitionMap.get(transition.getState()).removeIf(t -> t == transition);
```
See other comments in the file.
